c/src/ml-api-service.c

```diff
@@ -328,10 +458,8 @@ ml_service_new (const char *config, ml_service_h * handle)
 
   object = json_node_get_object (root);
 
-  if (json_object_has_member (object, "single") ||
-      json_object_has_member (object, "pipeline")) {
-    service_type = ML_SERVICE_TYPE_EXTENSION;
-  } else {
+  service_type = _ml_service_get_type (object);
+  if (ML_SERVICE_TYPE_UNKNOWN == service_type) {
```

Comment:
duplicated error message (line 398 and 463)

Reply:
Thanks, Updated!